libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/LanguagePolicy.cs

```diff
@@ -42,16 +58,12 @@ public class LanguagePolicy : Dictionary<string, string[]>
                     }
                 }
 
-                fallback.Add(string.Empty);
+                fallback.Add(defaultLanguage);
+
                 policy.Add(language, fallback.ToArray());
             }
 
             return policy;
-        });
-
-        public LanguagePolicy()
-            : base(defaultPolicy.Value, StringComparer.OrdinalIgnoreCase)
-        {
         }
     }
```

Comment:
I think we want to do this:
public class LanguagePolicy : Dictionary<string, List<string>>
{
 // Keep this method for JSON deserialization 
 public LanguagePolicy()
  : base(DefaultPolicy(), StringComparer.OrdinalIgnoreCase)
 {
 }

 public LanguagePolicy(string defaultLanguage)
  : base(DefaultPolicy(defaultLanguage), StringComparer.OrdinalIgnoreCase)
 {
 }

 // walk through all of the cultures and create a dictionary map with most specific to least specific
 // Example output "en-us" will generate fallback rule like this:
 //   "en-us" -> "en" -> "" 
 //   "en" -> ""
 // So that when we get a locale such as en-gb, we can try to resolve to "en-gb" then "en" then ""
 // See commented section for full sample of output of this function
 private static IDictionary<string, List<string>> DefaultPolicy(string defaultLanguage = null)
 {
  var cultureCodes = CultureInfo.GetCultures(CultureTypes.AllCultures).Select(c => c.IetfLanguageTag.ToLower()).ToList();
  var policy = new Dictionary<string, List<string>>(StringComparer.OrdinalIgnoreCase);
  foreach (var language in cultureCodes.Distinct())
  {
   var lang = language.ToLower();
   var fallback = new List<string>();
   while (!string.IsNullOrEmpty(lang))
   {
    fallback.Add(lang);

    var i = lang.LastIndexOf("-");
    if (i > 0)
    {
     lang = lang.Substring(0, i);
    }
    else
    {
     break;
    }
   }
   
   if (language != String.Empty)
   {
    policy.Add(language, fallback);
   }
  }
  
  if (defaultLanguage != null && policy.TryGetValue(defaultLanguage, out List<string> defaultPolicy))
  {
   foreach(var key in policy.Keys)
   {
    policy[key].AddRange(defaultPolicy);
   }
  }

  return policy;
 }
}


For new LanagueGenerator() This generates:
"de-de" => "de-de","de"

for new LanguageGeneartor("en-us") this generates:
"de-de" => "de-de","de","en-us","en"

This means full defaultPolicy for defaultLanguage (en-us) is appended to every language, if "en-us" is complex it captures that definition fully without any extra code on consumption side.

Reply:
I'm completely lost, 

> For new LanagueGenerator() This generates:
> "de-de" => "de-de","de"

this don't include "" anymore, and if we don't change our lookup logic, if will stop at search for de, and fail,  are you suggesting for now on, we don't support foo.lg anymore? 

> for new LanguageGeneartor("en-us") this generates:
> "de-de" => "de-de","de","en-us","en"

by this, are you suggesting we don't other approach to customize policy, and only let user pick a default language from what we defined? because i feels this method has a little much implicit assumption and rules here. 

What i would image is 
* for simple case, user just call new LanguagePolicy() or new LanguagePolicy("en-us"). 
* for advanced case, user call new LangugePolicy(dict) with a dict they created, or new LanguagePolc(), then update the policy like policy["en-us"] = ["my-en", "your en", "whatever"]. 

For whatever approach, i feel like keep the policy straightforward and easy to explain is more easier and extensible than try to list  scenarios